src/app/components/post/view/view.ts

```diff
@@ -81,4 +96,68 @@ export default class AppPostView extends Vue {
 			message: this.$gettext('Your post has been published.'),
 		});
 	}
+
+	mounted() {
+		this.listenGrid();
+	}
+
+	private listenGrid() {
+		// Check every 2 seconds if we are connected to grid.
+		// Grid bootstraps async to the site load, so this page might be mounted before grid finished loading.
+		// Once connected, start listening to the post's channel.
+		this.startListenInterval = setInterval(() => {
+			if (this.startedListeningGrid) {
+				if (this.startListenInterval) {
+					clearInterval(this.startListenInterval);
+					this.startListenInterval = null;
+				}
+				return;
+			}
+
+			if (this.grid && this.grid.connected) {
+				this.startedListeningGrid = true;
+				this.grid.startListenPost(this.post, this.handleGridEvent);
+			}
+		}, 2000);
+	}
+
+	handleGridEvent(event: GridEvent, payload: GridEventPayload) {
+		// If the logged in user has taken an action, don't update:
+		// That change is already visible (like++, poll vote etc)
+		if (this.app.user && this.app.user.id === payload.user_id) {
+			return;
+		}
+
+		switch (event) {
+			case 'poll-vote':
+				if (this.post.poll && payload.poll_item_id) {
+					const item = this.post.poll.items.find(i => i.id === payload.poll_item_id);
+					if (item) {
+						item.vote_count++;
+						this.post.poll.vote_count++;
+					}
+				}
+
+				break;
+
+			case 'like':
+				this.post.like_count++;
+				break;
+
+			case 'unlike':
+				this.post.like_count--;
+				break;
+		}
+	}
+
+	destroyed() {
```

Comment:
Check if it actually connected to the post. Also, did you check navigating between different posts?